homework/list.js

```diff
@@ -1,79 +1,85 @@
 'use strict';
 
-(function() {
+function List() {
+	this._listNoteContent = [];
+	this._wrapper = document.querySelector('#note-list-wrapper');
+}
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+List.prototype = {
+	start : function () {
+		this.fetchList().then(function (data){
+			this.updateList(data);
+		}.bind(this)).then(
+			this.drawList.bind(this)
+		).then(
+			this.preloadFirstNote.bind(this)
+		);
+		window.addEventListener('click', (function (event) {
+			this.onNoteOpen(event);
+		}).bind(this));
+	},
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
+	onNoteOpen : function (event) {
+		if (event.target.classList.contains('note-title')) {
+			var id = event.target.dataset.noteId;
+			var content = this._listNoteContent[id];
+			window.dispatchEvent(new CustomEvent('note-open',
+			{ detail: content }));
+		};
+	},
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
+	preloadFirstNote : function () {
+		if (this._listNoteContent.length !== 0) {
```

Comment:
nit: It's better to put the condition outside a function wholly wrapped by only one branch.